### PR TITLE
tikv-configuration: mark raft-engine.bytes-per-sync as deprecated

### PR DESCRIPTION
**Pull request by qiancai**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [x] v8.3 (TiDB 8.3 versions)
- [x] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by hbisheng — APPROVED** [review]
cc @LykxSassinator

File: tikv-configuration-file.md [hunk 1]
```diff
@@ -1957,9 +1957,13 @@ Raft Engine 相关的配置项。
...
 ### `bytes-per-sync`
 
+> **警告：**
+>
+> 从 v6.5.0 起，Raft Engine 在写入日志时不会缓存而是直接落盘，因此该配置项被废弃，且不再生效。
+
 + 指定缓存写入的最大累积大小。当超过此配置值时，缓存的写入将被刷写到磁盘。
 + 如果将此配置项设置为 `0`，则禁用增量同步。
-+ 默认值：`"4MiB"`
++ 默认值：`NULL`。在 v6.5.0 之前的版本中，默认值为 `"4MiB"`。
```
Comment (LykxSassinator) [inline tikv-configuration-file.md:1966]: 建议
```suggestion
+ 默认值：`0`。在 v6.5.0 之前的版本中，默认值为 `"4MiB"`。
```

不建议用`NULL` 来制定

Comment (hbisheng) [reply tikv-configuration-file.md:1966]: 好像在实际测试中读出来的就是 NULL。
img_v3_02g3_3f05f670-c6e0-482c-9f80-1a31a9927ddg

这里是不是关系不大，因为我看代码里这个参数已经不再起作用了。或者干脆就不要提默认值吗？@LykxSassinator

Comment (LykxSassinator) [reply tikv-configuration-file.md:1966]: 这个是 SQL 返回的 =》or 直接不提默认值也是ok的。

File: releases/release-6.5.0.md [hunk 2]
```diff
@@ -353,6 +353,7 @@ TiDB 6.5.0 为长期支持版本 (Long-Term Support Release, LTS)。
 | TiDB | [`server-memory-quota`](/tidb-configuration-file.md#server-memory-quota-从-v409-版本开始引入) | 废弃 | 自 v6.5.0 起，该配置项被废弃。请使用 [`tidb_server_memory_limit`](/system-variables.md#tidb_server_memory_limit-从-v640-版本开始引入) 系统变量进行设置。 |
 | TiDB | [`disconnect-on-expired-password`](/tidb-configuration-file.md#disconnect-on-expired-password-从-v650-版本开始引入) | 新增 | 该配置用于控制 TiDB 服务端是否直接断开密码已过期用户的连接，默认值为 `true`，表示 TiDB 服务端将直接断开密码已过期用户的连接。 |
 | TiKV | `raw-min-ts-outlier-threshold` | 删除 | 从 v6.4.0 起，该配置项被废弃。从 v6.5.0 起，该配置项被删除。 |
+| TiKV | [`raft-engine.bytes-per-sync`](/tikv-configuration-file.md#bytes-per-sync-2) | 废弃 | 从 v6.5.0 起，Raft Engine 在写入日志时不会缓存而是直接落盘，因此该配置项被废弃，且不再生效。默认值从 `"4MiB"` 修改为 `NULL`。 |
```
Comment (qiancai) [inline releases/release-6.5.0.md:356]:
```suggestion
| TiKV | [`raft-engine.bytes-per-sync`](/tikv-configuration-file.md#bytes-per-sync-2) | 废弃 | 从 v6.5.0 起，Raft Engine 在写入日志时不会缓存而是直接落盘，因此该配置项被废弃，且不再生效。 |
```

**Pull request merged**